src/delphi/dataset/token_map.py

```diff
@@ -0,0 +1,25 @@
+from typing import cast
+
+from datasets import Dataset
+
+
+def token_map(
+    tokenized_dataset: Dataset,
+) -> dict[int, list[tuple[int, int]]]:
+    """Return a mapping of tokens to their (prompt_idx, token_idx) locations in the tokenized_dataset.
+
+    Args:
+        tokenized_dataset (Dataset): A tokenized dataset.
+
+    Returns:
+        dict[int, list[tuple[int, int]]]: A mapping of tokens to their (prompt_idx, token_idx)
+            locations in the tokenized_dataset.
+    """
+    mapping = {}
+    tokenized_dataset = cast(Dataset, tokenized_dataset)
```

Comment:
Why cast here? What happens if you remove it?

Reply:
I initially cast in hope of getting pyright to realize prompt has a get_item, but the problem is elsewhere. Deleting this line seems to not affect the function in any way, so I'll remove it